### PR TITLE
[TAS-1626] 👽️ Use new ebook-cors API for reader

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

File: src/pages/reader/pdf.vue [hunk 1]
```diff
@@ -75,28 +87,30 @@ export default {
         }
       }
     },
+    async getFileBuffer() {
+      let buffer;
+      if (window.caches) {
+        try {
+          const cache = await caches.open('reader-epub');
```
Comment (nwingt) [inline src/pages/reader/pdf.vue:94]: PDF uses `reader-epub` too, how about a generic name for the cache? e.g. `reader-ebook`?

File: src/pages/reader/pdf.vue [hunk 2]
```diff
@@ -75,28 +87,30 @@ export default {
...
+          console.error(error);
+        }
+      }
+      if (!buffer) {
+        buffer = await this.$axios.$get(this.corsUrl, {
+          responseType: 'arraybuffer',
+        });
+      }
+      return buffer;
+    },
```
Comment (nwingt) [inline src/pages/reader/pdf.vue:109]: Refactor with epub?

File: src/pages/reader/pdf.vue (hunk 1 above)
Comment (williamchong) [reply src/pages/reader/pdf.vue:94]: actually it is suppose to be `reader-pdf`...

File: src/pages/reader/pdf.vue [hunk 3]
```diff
@@ -18,16 +18,29 @@
...
     },
+    corsUrl: {
+      type: String,
+      default: '',
+    },
+    cacheKey: {
+      type: String,
+      default: '',
+    },
   },
```
Comment (nwingt) [inline src/pages/reader/pdf.vue:44]: Props can be moved to reader mixin too

**Review by nwingt — APPROVED** [review]

**Pull request merged**